packages/control-group/src/styles.tsx

```diff
@@ -0,0 +1,274 @@
+import {Button as SmashingButton, getButtonStyle} from '@smashing/button'
+import styled, {DefaultTheme} from 'styled-components'
+import {WrapperProps, ButtonProps, Layout, TextAlign} from './types'
+
+interface ThemeProps {
+  theme: DefaultTheme
+}
+
+const getTextAlign = (textAlign?: TextAlign) => {
+  switch (textAlign) {
+    case 'center':
+      return 'center'
+    case 'right':
+      return 'flex-end'
+    default:
+      return 'flex-start'
+  }
+}
+const getTextColor = (invalid?: boolean, disabled?: boolean) => (
+  _: ThemeProps
+) => {
+  if (invalid) {
+    return _.theme.colors.text.danger
+  }
+  if (disabled) {
+    return _.theme.colors.text.muted
+  }
+
+  return _.theme.colors.text.intense
+}
+
+const getDisplay = (layout?: Layout) => {
+  switch (layout) {
+    case 'full':
+      return 'grid'
+    case 'equal':
+    default:
+      return 'inline-grid'
+  }
+}
+
+const getState = (_: ButtonProps & ThemeProps) => {
+  let state = 'default'
+
+  if (_.activeGroup || _.checked) {
+    state = 'active'
```

Comment:
```suggestion
    return 'active'
```

Reply:
I don't want to return 'active' right away, because if there is invalid/disabled it should override the active. But I can just change the order of ifs ;)

Reply:
fixed